src/libpmemobj/memblock.c

```diff
@@ -471,10 +472,15 @@ run_get_bitmap(const struct memory_block *m, struct run_bitmap *b)
 	struct chunk_header *hdr = heap_get_chunk_hdr(m->heap, m);
 	struct chunk_run *run = heap_get_chunk_run(m->heap, m);
 
-	uint32_t size_idx = hdr->size_idx;
-	memblock_run_bitmap(&size_idx, hdr->flags, run->hdr.block_size,
-		run->hdr.alignment, run->content, b);
-	ASSERTeq(size_idx, hdr->size_idx);
+	if (m->cached_bitmap != NULL) {
+		*b = *m->cached_bitmap;
+		b->values = (uint64_t *)run->content;
+	} else {
+		uint32_t size_idx = hdr->size_idx;
```

Comment:
can we store bitmap here to the cache it in the memory block?

Reply:
The cache is a pointer to a `run_bitmap` structure, and here we are simply operating on a stack-allocated variable. Your suggestion would require this code to do a malloc() and then add some extra logic for freeing that later on - which would be complicated.

tl;dr: no.